modules/ROOT/pages/authorization-grant-types.adoc

```diff
@@ -0,0 +1,103 @@
+= OAuth Authorization Grant Types
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: http, authentication, security, users, connectors, anypoint, studio, oauth
+
+There are four types of authorization grants that an OAuth consumer (that is, a client app) can use to obtain access to a protected resource from an OAuth service provider:
+
+* <<Authorization Code>>
+
+* <<Implicit>>
+
+* <<Resource Owner Password Credentials>>
+
+* <<Client Credentials>>
+
+For information about the OAuth2 RFC, see https://tools.ietf.org/html/rfc6749.
+
+////
+For more information about OAuth roles and the OAuth dance, see the xref:mule-secure-token-service.adoc[Mule Secure Token Service] topic.
+////
+
+== Authorization Code
+
+The Authorization Code grant type uses an authorization server (responsible for confirming and granting permission to access the protected resource) and a resource server (responsible for providing access to the protected resource). Though described as independent servers, the authorization and resource servers reside on the same Mule server. With the Authorization Code grant type, a client app must ask for an authorization code from the service provider before it can ask for the token that will grant access to protected resources.
+
+The Authorization Code grant type is especially secure, because it authenticates the client and transmits the access token without exposing it to unauthorized parties, including the resource owner.
+
+An OAuth dance that uses an Authorization Code grant type works as follows:
+
+. Client app wants to use a protected resource housed by the service provider, so the client initiates a request to access the resource.
+
+. Client app calls the service provider's authorization server, providing its client ID to confirm that it is a registered client of the service provider.
+
+. Client app uses a login page supplied by the service provider to authenticate itself to the service provider.
+
+. After confirming that the client ID is valid (that is, the client app had previously registered with the service provider), and authenticating the resource owner's login credentials, the service provider returns an authorization code.
+
+. Client app calls the authorization server again, and provides its authorization code, client ID (again), and client secret.
```

Comment:
its authorization code -> the authorization code

Reply:
OK